airflow/providers/google/cloud/triggers/bigquery_dts.py

```diff
@@ -0,0 +1,156 @@
+    account from the list granting this role to the originating account (templated).
+    """
+
+    def __init__(
+        self,
+        project_id: str,
+        config_id: str,
+        run_id: str,
+        poll_interval: int = 10,
+        gcp_conn_id: str = "google_cloud_default",
+        delegate_to: str | None = None,
+        location: str | None = None,
+        impersonation_chain: str | Sequence[str] | None = None,
+    ):
+        super().__init__()
+        self.project_id = project_id
+        self.config_id = config_id
+        self.run_id = run_id
+        self.poll_interval = poll_interval
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+        self.location = location
+        self.impersonation_chain = impersonation_chain
+
+    def serialize(self) -> tuple[str, dict[str, Any]]:
+        """Serializes class arguments and classpath."""
+        return (
+            "airflow.providers.google.cloud.triggers.bigquery_dts.BigQueryDataTransferRunTrigger",
+            {
+                "project_id": self.project_id,
+                "config_id": self.config_id,
+                "run_id": self.run_id,
+                "poll_interval": self.poll_interval,
+                "gcp_conn_id": self.gcp_conn_id,
+                "delegate_to": self.delegate_to,
+                "location": self.location,
+                "impersonation_chain": self.impersonation_chain,
+            },
+        )
+
+    async def run(self) -> AsyncIterator[TriggerEvent]:
+        """
+        Get Transfer Run status and if it one of the statuses which mean end of the job
+        then yield TriggerEvent object.
+        """
+        hook = self._get_async_hook()
+        while True:
+            try:
+                transfer_run: TransferRun = await hook.get_transfer_run(
+                    project_id=self.project_id,
+                    config_id=self.config_id,
+                    run_id=self.run_id,
+                )
+                state = transfer_run.state
+                self.log.info("Current state is %s", state)
+
+                if state == TransferState.SUCCEEDED:
+                    self.log.info("Job has completed it's work.")
+                    yield TriggerEvent(
+                        {
+                            "status": "success",
+                            "run_id": self.run_id,
+                            "message": "Job completed",
+                        }
+                    )
+                    return
+
+                elif state == TransferState.FAILED:
+                    self.log.info("Job has failed")
+                    yield TriggerEvent(
+                        {
+                            "status": "failed",
+                            "run_id": self.run_id,
+                            "message": "Job has failed",
+                        }
+                    )
+                    return
```

Comment:
```suggestion
```

Reply:
Please check my response about returning statement after yield in Trigger class